x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/index.tsx

```diff
@@ -7,6 +7,13 @@
 
 import { lazy } from 'react';
 import { suspendedComponentWithProps } from '../../lib/suspended_component_with_props';
+import type { RuleAddComponent } from './rule_add';
+import type { RuleEditComponent } from './rule_edit';
 
-export const RuleAdd = suspendedComponentWithProps(lazy(() => import('./rule_add')));
-export const RuleEdit = suspendedComponentWithProps(lazy(() => import('./rule_edit')));
+export const RuleAdd = suspendedComponentWithProps(
+  lazy(() => import('./rule_add'))
+) as RuleAddComponent; // `React.lazy` is not typed correctly to support generics so casting back to imported component
```

Comment:
would `dynamic` from `@kbn/shared-ux-utility` help in this case?

Reply:
Same for `RuleEdit`

Reply:
Has the same issues unfortunately :/